core/services/editor/EditorBlockAssetRegister.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace EventEspresso\core\services\editor;
+
+use EventEspresso\core\domain\DomainInterface;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\services\assets\AssetRegisterInterface;
+use EventEspresso\core\services\assets\Registry;
+
+defined('EVENT_ESPRESSO_VERSION') || exit;
+
+
+
+/**
+ * Class EditorBlockAssetRegister
+ * Abstract parent for classes that handle asset registration for one or more editor blocks that share the same assets
+ *
+ * @package EventEspresso\core\services\editor
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+abstract class EditorBlockAssetRegister implements AssetRegisterInterface
+{
+
+    /**
+     * @var DomainInterface $domain
+     */
+    private $domain;
+
+    /**
+     * @var Registry $registry
+     */
+    private $registry;
+
+    /**
+     * @var string $script_handle
+     */
+    private $script_handle;
+
+    /**
+     * @var array $script_dependencies
+     */
+    private $script_dependencies;
+
+    /**
+     * @var string $style_handle
+     */
+    private $style_handle;
+
+    /**
+     * @var array $style_dependencies
+     */
+    private $style_dependencies;
+
+
+    /**
+     * EditorBlockAssetRegister constructor.
+     *
+     * @param string          $script_handle
+     * @param array           $script_dependencies
+     * @param string          $style_handle
+     * @param array           $style_dependencies
+     * @param DomainInterface $domain
+     * @param Registry        $registry
+     */
+    public function __construct(
+        $script_handle,
+        array $script_dependencies,
+        $style_handle,
+        array $style_dependencies,
+        DomainInterface $domain,
+        Registry $registry
+    ) {
+        $this->script_handle       = $script_handle;
+        $this->script_dependencies = $script_dependencies;
+        $this->style_handle        = $style_handle;
+        $this->style_dependencies  = $style_dependencies;
+        $this->domain              = $domain;
+        $this->registry            = $registry;
+    }
```

Comment:
**Absolutely No**!!!

That is a bad idea that should be avoided like the plague, so please don't do this kind of thing anywhere, and I'll explain why.

A quick example that hopefully makes it instantly clear is "unit testing". 

The longer answer is that a class should directly declares its dependencies by type hinting for them in it's constructor, which allows other classes to know what it requires in order to operate. Obtaining dependencies via other objects, hides a class's dependencies and is a form of "service location", which is discouraged for many reasons (you can read more about that if you want to by Googling that term). This also binds both of those classes to each other and prevents them from being able to change if/when their responsibilities change. 

For example: 

 - class A requires both class B and C in order to function and is therefore dependent on both of those classes
 - class B is also dependent on class C and already has that dependency being injected into it
 - class A indirectly obtains class C from class B (so class A is **only** injecting class B) instead of requiring it directly
 - a new business requirement comes along (like GDPR) which forces changes to existing domain logic. 
 - what if the logic in class B that requires class C needs to be extracted into a new class (class D)? Class B would no longer have an instance of class C available to pass along to class A. Now class A needs to be refactored in order for it to obtain an instance of class C which it can no longer retrieve from class B.
 - or what if it is determined that class C needs to be split into multiple subclasses (like D & E) that extend class C, and class B's requirement changes but class A's does not? So class B now requires class D (subclass of C), and class A now requires class E (also a subclass of C). This issue might be harder to spot because Class A can still obtain an object from class B that appears to be an instance of class C, but is actually now an instance of class D, however the data and/or functionality that class A requires has now been moved into class E. 

So what this means is that class B is either forced to maintain dependencies that it no longer requires (bloat), or class A is forced to change solely because of changes required in class B (the domino effect).

Going back to the simple example of "unit testing" as a reason for not hiding dependencies, and we can see that service location obviously interferes with the ability to inject mocks into a class for the purpose of testing it. If a class is obtaining its dependencies from another class, then any injected mock now has to also inject a mock of the second class. This of course is not impossible but should immediately highlight how these classes have become tangled up together. It's even worse if that service location or injection is happening via static method calls because now the class providing the dependency can not even be substituted for a mock. This is a common problem in EE right now because of our past usage of singletons.

So bottom line: Injecting a class's dependencies directly allows them to be controlled externally at runtime, which can then be done via a DI container, or a unit testcase setup method, etc. and gives classes the freedom to change when required and also to not have changes forced upon them when not required. This greatly enhances the solidity of code because it better adheres to the Single Responsibility Principle.